### PR TITLE
integration-tests/smoke: add plugins variant to TestOCRv2Basic

### PR DESCRIPTION
**Pull request by jmank88**

### CODE REVIEW

**Review by ilija42 — COMMENTED** [review]
Does this affect CI? Not completely sure how these smoke tests are ran, IIUC it uses this same smoke test for plugin and non plugin variant, but just builds it differently?

**Comment by jmank88** [comment]
> Does this affect CI? Not completely sure how these smoke tests are ran, IIUC it uses this same smoke test for plugin and non plugin variant, but just builds it differently?

Yes, you can see the subtest in the logs for the ocr2-* jobs. There is a `-plugins` variant already, but that is running a different dockerfile, which has all plugins, and enables them by default. This test will execute on the regular docker file too, which has limited plugins, and disables them by default. This way we try the feature flag both ways on both images.

It does run more than necessary though, since there are so many ocr2-* variants for eth clients, so maybe there is a better way to organize this. :thinking: 
cc @kalverra

**Comment by jmank88** [comment]
> These are new additions that I'm currently [working on moving to a nightly job](https://github.com/smartcontractkit/chainlink/pull/11610) instead of every PR. If you feel the plugins tests need a similar treatment we can move them to that style as well.

It is important to run on every PR, but only once. So maybe this is fine as-is, but could wait for that PR first?
 
> I would prefer to make it a little more explicit what this is meant to do regarding running on a regular image with feature flags shut off. Maybe make a comment along with renaming the sub test to `with-disabled-plugins` or something?

Can you elaborate?
The sub-tests are named `legacy` and `plugins` which is a pattern we've established in a few places. The regular image has the feature disabled already, so `legacy` env vars are a no-op, but `plugins` validates that we can enable the feature. The `-plugins` image has it enabled by default, so `legacy` disables and `plugins` is a no-op.

(we will be able to drop the -plugins image eventually - hopefully in Q1)

**Comment by kalverra** [comment]
> It is important to run on every PR, but only once. So maybe this is fine as-is, but could wait for that PR first?

By "it" here, do you mean the plugins tests? Or the client tests?

> (we will be able to drop the -plugins image eventually - hopefully in Q1)

Ah, okay, never mind then.

**Review by nolag — DISMISSED** [review]

**Comment by kalverra** [comment]
> Clients are less significant, although we could probably trigger it for certain chains/evm/clients paths or whatever is related

We're on the same page then. I'll keep it to just nightly for now, unless you have some specific paths in mind at the moment.

**Review by ilija42 — APPROVED** [review]

File: integration-tests/smoke/ocr2_test.go [hunk 1]
```diff
@@ -16,86 +16,136 @@ import (
...
+      require.NoError(t, err)
+      err = actions.WatchNewOCR2Round(2, aggregatorContracts, env.EVMClient, time.Minute*5, l)
+      require.NoError(t, err)
+
+      roundData, err = aggregatorContracts[0].GetRound(testcontext.Get(t), big.NewInt(2))
+      require.NoError(t, err, "Error getting latest OCR answer")
+      require.Equal(t, int64(10), roundData.Answer.Int64(),
+        "Expected latest answer from OCR contract to be 10 but got %d",
+        roundData.Answer.Int64(),
+      )
```
Comment (jmank88) [inline integration-tests/smoke/ocr2_test.go:146]:
```suggestion
			ocrOffchainOptions := contracts.DefaultOffChainAggregatorOptions()
			aggregatorContracts, err := actions.DeployOCRv2Contracts(1, linkToken, env.ContractDeployer, transmitters, env.EVMClient, ocrOffchainOptions)
			require.NoError(t, err, "Error deploying OCRv2 aggregator contracts")
			
			err = actions.CreateOCRv2JobsLocal(aggregatorContracts, bootstrapNode, workerNodes, env.MockAdapter, "ocr2", 5, env.EVMClient.GetChainID().Uint64(), false)
			require.NoError(t, err, "Error creating OCRv2 jobs")
			
			ocrv2Config, err := actions.BuildMedianOCR2ConfigLocal(workerNodes, ocrOffchainOptions)
			require.NoError(t, err, "Error building OCRv2 config")
			
			err = actions.ConfigureOCRv2AggregatorContracts(env.EVMClient, ocrv2Config, aggregatorContracts)
			require.NoError(t, err, "Error configuring OCRv2 aggregator contracts")
			
			err = actions.WatchNewOCR2Round(1, aggregatorContracts, env.EVMClient, time.Minute*5, l)
			require.NoError(t, err, "Error starting new OCR2 round")
			roundData, err := aggregatorContracts[0].GetRound(testcontext.Get(t), big.NewInt(1))
			require.NoError(t, err, "Getting latest answer from OCR contract shouldn't fail")
			require.Equal(t, int64(5), roundData.Answer.Int64(),
			  "Expected latest answer from OCR contract to be 5 but got %d",
			  roundData.Answer.Int64(),
			)
			
			err = env.MockAdapter.SetAdapterBasedIntValuePath("ocr2", []string{http.MethodGet, http.MethodPost}, 10)
			require.NoError(t, err)
			err = actions.WatchNewOCR2Round(2, aggregatorContracts, env.EVMClient, time.Minute*5, l)
			require.NoError(t, err)
			
			roundData, err = aggregatorContracts[0].GetRound(testcontext.Get(t), big.NewInt(2))
			require.NoError(t, err, "Error getting latest OCR answer")
			require.Equal(t, int64(10), roundData.Answer.Int64(),
			  "Expected latest answer from OCR contract to be 10 but got %d",
			  roundData.Answer.Int64(),
			)
```

**Comment by jmank88** [comment]
Rebased since merge conflicts were a mess.

**Review by kalverra — APPROVED** [review]

**Pull request merged**